services/asset_bundle/zip_asset_bundle.cc

```diff
@@ -21,19 +21,47 @@ void Ignored(bool) {
 
 }  // namespace
 
-ZipAssetBundle* ZipAssetBundle::Create(
+unzFile ScopedUnzFileTraits::InvalidValue() {
+  return nullptr;
+}
+
+void ScopedUnzFileTraits::Free(unzFile file) {
+  unzClose(file);
+}
+
+scoped_refptr<ZipAssetBundle> ZipAssetService::Create(
     InterfaceRequest<AssetBundle> request,
     const base::FilePath& zip_path,
     scoped_refptr<base::TaskRunner> worker_runner) {
-  return new ZipAssetBundle(request.Pass(), zip_path, worker_runner.Pass());
+  scoped_refptr<ZipAssetBundle> zip_asset_bundle(
+      new ZipAssetBundle(zip_path, worker_runner.Pass()));
+
+  // Register as a Mojo service.
+  new ZipAssetService(request.Pass(), zip_asset_bundle);
```

Comment:
I probably would have make this a separate step that the client does by calling a method on the ZipAssetBundle.